apps/spruce/src/pages/version/Tabs/VersionTiming/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { useQueryParams } from "hooks/useQueryParam";
+import { PatchTasksQueryParams } from "types/task";
+import { parseQueryString } from "utils/queryString";
+import { useQueryVariables } from "../useQueryVariables";
+import {
+  transformTaskDurationDataToVariantGanttChartData,
+  transformTaskDurationDataToTaskGanttChartData,
+} from "./utils";
+import VersionTimingGraph from "./VersionTimingGraph";
+
+interface Props {
+  versionId: string;
+  taskCount: number;
+}
+
+const VersionTiming: React.FC<Props> = ({ taskCount, versionId }) => {
+  const dispatchToast = useToastContext();
+  const { search } = useLocation();
+
+  const defaultSort = `${TaskSortCategory.Duration}:${SortDirection.Desc}`;
+  const [queryParams, setQueryParams] = useQueryParams();
+  const versionAnalytics = useVersionAnalytics(versionId);
+  const queryVariables = useQueryVariables(search, versionId);
+  const hasQueryVariables = Object.keys(parseQueryString(search)).length > 0;
+  const hasBuildVariantQueryParam = !!queryParams.variant;
+  const shouldPaginateData = hasBuildVariantQueryParam;
+
+  useEffect(() => {
+    setQueryParams({
+      ...queryParams,
+      [TableQueryParams.Sorts]: defaultSort,
+    });
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const clearQueryParams = () => {
+    setQueryParams({
+      ...queryParams,
+      [PatchTasksQueryParams.TaskName]: undefined,
+      [PatchTasksQueryParams.Variant]: undefined,
+      [PatchTasksQueryParams.Statuses]: undefined,
+      [PatchTasksQueryParams.BaseStatuses]: undefined,
+      [PaginationQueryParams.Page]: undefined,
+      [TableQueryParams.Sorts]: defaultSort,
+    });
+  };
+
+  const { data, loading, refetch, startPolling, stopPolling } = useQuery<
+    VersionTaskDurationsQuery,
+    VersionTaskDurationsQueryVariables
+  >(VERSION_TASK_DURATIONS, {
+    variables: {
+      ...queryVariables,
+      taskFilterOptions: {
+        ...queryVariables.taskFilterOptions,
+        limit:
+          shouldPaginateData && queryVariables.taskFilterOptions.limit
+            ? queryVariables.taskFilterOptions.limit
+            : undefined,
+      },
+    },
+    skip: !hasQueryVariables,
+    pollInterval: DEFAULT_POLL_INTERVAL,
+    onError: (err) => {
+      dispatchToast.error(`Error fetching patch tasks ${err}`);
+    },
+  });
+  usePolling({ startPolling, stopPolling, refetch });
+  const { version } = data || {};
+  const { tasks } = version || {};
+  const { count = 0, data: tasksData = [] } = tasks || {};
+  const { limit, page } = queryVariables.taskFilterOptions;
+
+  const chartData = hasBuildVariantQueryParam
+    ? transformTaskDurationDataToTaskGanttChartData(tasksData)
+    : transformTaskDurationDataToVariantGanttChartData(tasksData);
+
+  return (
+    <>
+      <Body>
+        This page provides a timeline view of{" "}
+        {hasBuildVariantQueryParam ? "task" : "variant"} wall clock duration in
+        the version.
+      </Body>
```

Comment:
How do you feel about this description instead
```suggestion
  const description = isVariantTimingView ? (
    <>
      This page is showing a timeline view of task run times in the{" "}
      <b>{queryParams.variant}</b> variant. This is a Gantt chart showing when
      each task started and finished running. You can click on a task to visit
      the task page.
    </>
  ) : (
    "This page is showing a timeline view of variant run times in this version. This is a Gantt chart showing when each variant started and finished running. You can click on a variant to see a view of the tasks that ran."
  );
  return (
    <>
      <H2>{isVariantTimingView ? "Variant" : "Version"} Timing</H2>
      <StyledBody>{description}</StyledBody>
```

When viewing the overall version and its variants
<img width="952" alt="image" src="https://github.com/user-attachments/assets/99b73829-6a32-4b42-80b1-89d7120992b8" />

When viewing a variant
<img width="968" alt="image" src="https://github.com/user-attachments/assets/3c703d10-f43b-4a67-812e-4cc9da5b7e88" />